os/arch/arm/src/armv7-m/mpu.h

```diff
@@ -672,7 +672,7 @@ static inline void up_set_mpu_app_configuration(struct tcb_s *rtcb)
 		return;
 	}
 
-#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+#if defined(CONFIG_OPTIMIZE_APP_RELOAD_TIME) || defined(CONFIG_MPU_STACKGUARD)
```

Comment:
Since, this is applicable to protected build also, its better to make a separate API for stack protection setting, and then call this new api from all the context switch places.

Reply:
changed the function for the mpu setting.